orbitize/system.py

```diff
@@ -144,245 +122,395 @@ def __init__(self, num_secondary_bodies, data_table, stellar_mass,
                                      ((len(self.radec[1]) + len(self.seppa[1]) + len(self.rv[1]) < len(data_table)) or \
                                       (self.num_secondary_bodies > 1))
 
+        if self.hipparcos_IAD is not None:
+            self.track_planet_perturbs = True
+
         if restrict_angle_ranges:
             angle_upperlim = np.pi
         else:
             angle_upperlim = 2.*np.pi
 
-        #
-        # Set priors for each orbital element
-        #
-
-        for body in np.arange(num_secondary_bodies):
-            # Add semimajor axis prior
-            self.sys_priors.append(priors.LogUniformPrior(0.001, 1e7))
-            self.labels.append('sma{}'.format(body+1))
-
-            # Add eccentricity prior
-            self.sys_priors.append(priors.UniformPrior(0., 1.))
-            self.labels.append('ecc{}'.format(body+1))
-
-            # Add inclination angle prior
-            self.sys_priors.append(priors.SinPrior())
-            self.labels.append('inc{}'.format(body+1))
-
-            # Add argument of periastron prior
-            self.sys_priors.append(priors.UniformPrior(0., 2.*np.pi))
-            self.labels.append('aop{}'.format(body+1))
-
-            # Add position angle of nodes prior
-            self.sys_priors.append(priors.UniformPrior(0., angle_upperlim))
-            self.labels.append('pan{}'.format(body+1))
-
-            # Add epoch of periastron prior.
-            self.sys_priors.append(priors.UniformPrior(0., 1.))
-            self.labels.append('tau{}'.format(body+1))
-
-        #
-        # Set priors on total mass and parallax
-        #
-        self.labels.append('plx')
-        if plx_err > 0:
-            self.sys_priors.append(priors.GaussianPrior(plx, plx_err))
-        else:
-            self.sys_priors.append(plx)
+        # Check for rv data
+        contains_rv = False
+        if len(self.rv[0]) > 0:
+            contains_rv = True
 
-        # checking for rv data to include appropriate rv priors:
+        # Assign priors for the given basis set
+        self.extra_basis_kwargs = {}
+        basis_obj = getattr(basis, self.fitting_basis)
 
-        if len(self.rv[0]) > 0 and self.fit_secondary_mass:
-            # Rob and Lea:
-            # for instrument in rv_instruments:
-                # add gamma and sigma for each and label each unique gamma and sigma per instrument name (gamma+instrument1, ...)
-            for instrument in self.rv_instruments:
-                self.sys_priors.append(priors.UniformPrior(-5, 5))  # gamma prior in km/s
-                self.labels.append('gamma_{}'.format(instrument))
+        # Obtain extra necessary data to assign priors for XYZ
+        if fitting_basis == 'XYZ':
```

Comment:
We should move this into the xyz class in the future. Can basically pass in the data table to the XYZ class to help it create priors.

Reply:
good call. made an issue. Issue #271 .